### DISCUSSION TITLE
Can I use Anthropic Claude through Vertex.AI by using FirebaseVertexAI?

### DISCUSSION THREAD

**Discussion by fumito-ito** [firebase/firebase-ios-sdk#13247] [Q&A] [answered]
Hi team !

I am creating an unofficial Swift SDK for Anthropic Claude.
https://github.com/fumito-ito/AnthropicSwiftSDK

As Anthropic Claude is deployed on several platforms, I am also developing extensions to easily call Claude on each platform.

Vertex.AI is one of the leading platforms supporting Anthropic Claude, but whether it is possible to use Anthropic Claude on VertexAI using `FirebaseVertexAI` or `[Google AI SDK for Swift](https://github.com/google-gemini/generative-ai-swift)` could not be deduced from the documentation.

Is it possible to use Anthropic Claude on VertexAI using `FirebaseVertexAI` or `Google AI SDK for Swift`? For example, I expect to be able to use the following code.

```swift
import FirebaseVertexAI

let vertex = VertexAI.vertexAI()
let model = vertex.generativeModel(modelName: "claude-3-sonnet@20240229")
```

**Comment by paulb777** [comment, marked as answer]
Both FirebaseVertexAI and Google AI SDK are opinionated about the models they work with so they can provide strongly typed Swift interfaces.  Thus it's unlikely that either would work with Anthropic Claude

**Reply by fumito-ito** [reply]
Thanks @paulb777 ! I would use the Vertex AI API directly.
